Code/Tools/AssetProcessor/native/ui/ProductDependencyTreeDelegate.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "ProductDependencyTreeDelegate.h"
+#include "ProductDependencyTreeItemData.h"
+#include "ProductAssetDetailsPanel.h"
+#include "ProductAssetTreeItemData.h"
+
+#include <QMouseEvent>
+#include <QApplication>
+#include <AzCore/Debug/Trace.h>
+#include <iostream>
+namespace AssetProcessor
+{
+    ProductDependencyTreeDelegate::ProductDependencyTreeDelegate(QObject* parent, ProductAssetDetailsPanel* panel)
```

Comment:
for some of these cases you can use a QPointer, they track their lifecycle so when the object is disposed of it becomes inaccessible.

Reply:
That's a good point. I changed `ProductAssetDetailsPanel*` into `QPointer<ProductAssetDetailsPanel>`.